client/my-sites/current-site/notice.jsx

```diff
@@ -92,7 +92,7 @@ const SiteNotice = React.createClass( {
 
 export default connect( ( state, ownProps ) => {
 	return {
-		hasDomainCredit: hasDomainCredit( state, ownProps.site.ID )
+		hasDomainCredit: !! ownProps.site && hasDomainCredit( state, ownProps.site.ID )
```

Comment:
Isn't the convention to use `Boolean()` instead of `!!`?

FWIW, we usually just check only if it's falsy:

```
  hasDomainCredit: ownProps.site && hasDomainCredit( state, ownProps.site.ID )
```